exasol/analytics/query_handler/graph/stage/sql/input_output.py

```diff
@@ -1,11 +1,12 @@
-import dataclasses
+from dataclasses import dataclass, field
+from typing import Dict, Protocol
 
 from exasol.analytics.query_handler.graph.stage.sql.dataset import Dataset
 from exasol.analytics.query_handler.graph.stage.sql.dependency import Dependencies
-from exasol.analytics.utils.data_classes_runtime_type_check import check_dataclass_types
+# from exasol.analytics.utils.data_classes_runtime_type_check import check_dataclass_types
 
 
-@dataclasses.dataclass(frozen=True, eq=True)
+@dataclass(frozen=True)
```

Comment:
Also, class `SQLStageInputOutput` has been simplified by leaving most implementation details up to the user. 
Again this made some tests obsolete and also required tests to instantiate `MultiDatasetSQLStageInputOutput` instead of the abstract parent class.